### PR TITLE
Fixes minor issues in CLI experience

### LINKED ISSUE #309: JSON is parsed twice when running `test` command
The first time the IEnumerable<JToken> for the test utterances is evaluated is here: https://github.com/microsoft/NLU.DevOps/blob/master/src/NLU.DevOps.CommandLine/EnumerableExtensions.cs#L21

We should memoize the JSON test cases so we don't parse it multiple times.

### PR DESCRIPTION
**Pull request by rozele**
Fixes issue where we parse JSON twice due to the lazily evaluated IEnumerable<JToken> used for enumerating test cases. It removes the need to use the `Enumerable.Count<T>()` extension method in the parallelized map operation to get test results.

This change also replaces occurences of `Task.Wait()` with `Task.GetAwaiter().GetResult()` to improve the stack trace for better error reporting.

Fixes #309

### CODE REVIEW

File: src/NLU.DevOps.CommandLine/EnumerableExtensions.cs [hunk 1]
```diff
@@ -38,6 +38,7 @@ internal static class EnumerableExtensions
                 }
 
                 tasks.Add(selectWithIndexAsync(indexedItem.Item, indexedItem.Index));
+                results.Add(default);
```
Comment (rozele) [inline src/NLU.DevOps.CommandLine/EnumerableExtensions.cs:41]: Add before async method to ensure we do not get an `IndexOutOfRangeException`

File: src/NLU.DevOps.CommandLine/NLU.DevOps.CommandLine.csproj [hunk 2]
```diff
@@ -16,8 +16,7 @@
     <TargetsForTfmSpecificContentInPackage>PublishProviders;$(TargetsForTfmSpecificContentInPackage)</TargetsForTfmSpecificContentInPackage>
   </PropertyGroup>
 
-  <Target Name="PublishProviders"
-          BeforeTargets="CoreBuild">
+  <Target Name="PublishProviders" BeforeTargets="CoreBuild">
```
Comment (rozele) [inline src/NLU.DevOps.CommandLine/NLU.DevOps.CommandLine.csproj:19]: Revert change.

File: src/NLU.DevOps.CommandLine/Program.cs [hunk 3]
```diff
@@ -13,7 +15,13 @@ namespace NLU.DevOps.CommandLine
...
...VdrQWR3QkdwVlhyZGZxUklEQzFvb2xwOVBOOUVTeEIKZEw3OVNiaUZxL0w4Y1A1dFJ3QUFBWENGVGQrZUFBQUVBd0JJTUVZQ0lRRFhyd3k3eWg4TW1mVU5p ckpNbWhsNQp0UE5wOHdPV21Wb2p3SjVxanpweW5BSWhBUGtXbHZKZFhnWGh6UXF6SUhxWVdZejNsTS9aMVpadU5YakRQU3FYCmh3Rm9BSFlBMzZWZXEyaUNUeDlzcmU2 NFgwNCtXdXJOb2hLa2FsNk9PeExBSUVSY0tuTUFBQUZ3aFUzZmJ3QUEKQkFNQVJ6QkZBaUVBb3R6UjhDdy9iOEZXUWtWODVJbHh1LzVnSktBa05UM01TTVliYzJvaU8r RUNJSG05Rng0UQoxR1ROOTBnR0RNeUZFckJPS053NXFZWmY2dzltOVdrSkNCanBBSFlBUWNqS3NkOGlSa29ReHFFNkNVS0hYazR4Cml4c0Q2K3RMeDJqd2tHS1dCdllB QUFGd2hVM2ZqQUFBQkFNQVJ6QkZBaUFjdVhYU3RZZndiTHI4NkZ1RTRiK2gKeDJ0RzN1aS9kUkZERFlITDA3N29KQUloQU0vMWo5N3BZN1N6WWszbVhnOW1DMEd2M2R3 TEpVYlhxVlkySUpNZQpPd21QTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFBYXIwV0w4VHU3WllRSHduRzVzczFuN1JqdGFGUnhvYVFGCm4xV2ZxdjFpVHdrQjNiczVo QUU2Sjk2aGh5WE1BbUVQUHUwR3BWVlJIRXc1UkN1YUNDNWpveVJ5RGtxVjRjTE0KR0Q5TVpPQW5Oa2duUzBuYVZJcFczSllwRHZpb2hTdWt1d0hSSFRsM1hsZkE4RFlr L1NadWV5TUZHVkFsVjkzWApWRjc4b0tqQjgwdUE5NzFnWDBsWlBZaEhNdGs1Y0huWjhiQVZaamw0WDZRY0FvRzBxeDlnTlZvVERLK1I0OHVJCkxybXRibjRRMkErYjhE TWpNMWhxU1VoWkcrSHBRM2ZxZGpZNzBhcWhMTjRvODNIY1I1S210MnpYU2Q5TkpTL0UKL0N1ZWdPS1Y0TGZ4dEl5MEdZVnovbCtlSmxmTUY2Q09lTndPdENPbFJnTDBU UXlJQzNIYwotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t"));
```
Comment (rozele) [inline src/NLU.DevOps.CommandLine/Program.cs:20]: Revert change.

File: src/NLU.DevOps.CommandLine/EnumerableExtensions.cs [hunk 4]
```diff
@@ -18,7 +18,7 @@ internal static class EnumerableExtensions
             }
 
             var indexedItems = items.Select((item, i) => new { Item = item, Index = i });
-            var results = new TResult[items.Count()];
+            var results = new List<TResult>();
```
Comment (AndyCW) [inline src/NLU.DevOps.CommandLine/EnumerableExtensions.cs:21]: Comment to self (discussed in IM): By eliminating usage of `Enumerable.Count<T>()` here, we remove the need to enumerate the full lazy list to get the number of tests.
The list is enumerated below in the `foreach` at line 30.
The function to determine the set of test cases includes logic to parse the JSON, so by eliminating the extraneous enumeration, we only parse the |JSON once now.

**Review by AndyCW — APPROVED** [review]
LGTM.

File: src/NLU.DevOps.CommandLine/EnumerableExtensions.cs (hunk 4 above)
Comment (rozele) [reply src/NLU.DevOps.CommandLine/EnumerableExtensions.cs:21]: 👍

**Pull request merged**